### PR TITLE
gpu count endpoint

### PR DESCRIPTION
**Pull request by dleviminzi**
This PR adds an endpoint that returns the platform's GPU count.

### CODE REVIEW

File: pkg/repository/base.go [hunk 1]
```diff
@@ -143,6 +143,7 @@ type ProviderRepository interface {
 	SetMachineLock(providerName, poolName, machineId string) error
 	RemoveMachineLock(providerName, poolName, machineId string) error
 	GetGPUAvailability(pools map[string]types.WorkerPoolConfig) (map[string]bool, error)
+	GetGPUCounts(pools map[string]types.WorkerPoolConfig) (map[string]int, error)
```
Comment (luke-lombardi) [inline pkg/repository/base.go:146]: looking at the function above this, do we really need a separate repository function for this?

can't we use the results from GetGPUAvailability (maybe with something else added), and use that in the new endpoint?

**Review by luke-lombardi — CHANGES_REQUESTED** [review]

File: pkg/repository/base.go (hunk 1 above)
Comment (dleviminzi) [reply pkg/repository/base.go:146]: `GetGPUAvailability` is just returning a map of GPU to boolean values and does not really count. If we just want to say there is no availability at all that could work, but it doesn't give us much flexibility to warn about low availability.

File: pkg/api/v1/machine.go [hunk 2]
```diff
@@ -28,6 +29,7 @@ func NewMachineGroup(g *echo.Group, providerRepo repository.ProviderRepository,
 		config:       config,
 	}
 
+	g.GET("/:workspaceId/gpus", auth.WithWorkspaceAuth(group.GPUCounts))
```
Comment (luke-lombardi) [inline pkg/api/v1/machine.go:32]: not sure if you mentioned this, but why do we need workspace Id here?

**Review by luke-lombardi — CHANGES_REQUESTED** [review]
one last comment on this one

File: pkg/api/v1/machine.go (hunk 2 above)
Comment (dleviminzi) [reply pkg/api/v1/machine.go:32]: > not sure if you mentioned this, but why do we need workspace Id here?

If I remember correctly, there was an issue with workspace auth if it was not included.

**Review by luke-lombardi — APPROVED** [review]

**Pull request merged**